modules/navier_stokes/src/fvkernels/PINSFVMomentumFriction.C

```diff
@@ -63,35 +102,71 @@ PINSFVMomentumFriction::PINSFVMomentumFriction(const InputParameters & params)
 //
 // \nabla p = \mu D v + \frac{\rho}{2} F |v| v
 //
-// where v denotes the interstitial velocity (e.g. the true fluid velocity). Note that to be
-// consistent with our overall porous Navier-Stokes formulation, we must multiply the above equation
-// by porosity:
-//
-// \epsilon \nabla p = \epsilon \mu D v + \epsilon \frac{\rho}{2} F |v| v =
-// \mu D v_D + \frac{\rho}{2} F |v| v_D
-//
-// where v_D is the superficial velocity, which is the variable this kernel is acting on. The two
-// terms on the final RHS are what this kernel implements. Because both monolithic and segregated
-// solves multiply by v_D in the gatherRCData and computeSegregatedContribution methods
-// respectively, it is the job of the computeFrictionWCoefficient method to compute, for the Darcy
-// term:
+// where v denotes the either the superficial or intersticial velocity and similarly for density
+// rho. Both monolithic and segregated solves get multiplied by v in the gatherRCData and
+// computeSegregatedContribution methods respectively, it is the job of the
+// computeFrictionWCoefficient method to compute, for the Darcy term:
 //
 // \mu D
 //
 // and for the Forchheimer term
 //
 // \frac{\rho}{2} F |v|
+//
+// For the non standard formulation we define :
+// \nabla p = D v + F |v| v
+//
+// For the intersticial version we need to define: v = v/_epsilon and rho = rho *_epsilon
 
 ADReal
 PINSFVMomentumFriction::computeFrictionWCoefficient(const Moose::ElemArg & elem_arg,
                                                     const Moose::StateArg & state)
 {
+  // Forward declaration of the coeffcient to be returned by the model
   ADReal coefficient = 0.0;
-  if (_use_Darcy_friction_model)
-    coefficient += (*_mu)(elem_arg, state) * (*_D)(elem_arg, state)(_index);
-  if (_use_Forchheimer_friction_model)
-    coefficient +=
-        _rho(elem_arg, state) / 2 * (*_F)(elem_arg, state)(_index) * (*_speed)(elem_arg, state);
+  ADReal speed = 0.0;
+  ADReal rho = 0.0;
+  if (_use_Forchheimer_friction_model && _speed)
+    speed = (*_speed)(elem_arg, state);
+  else if (_use_Forchheimer_friction_model && !_speed)
+  {
+    ADRealVectorValue superficial_velocity;
+    if (_dim >= 1)
+    {
+      superficial_velocity(0) = (*_u_var)(elem_arg, state);
+      if (_dim >= 2)
+      {
+        superficial_velocity(1) = (*_v_var)(elem_arg, state);
+        if (_dim >= 3)
+          superficial_velocity(2) = (*_w_var)(elem_arg, state);
+      }
+    }
+    speed = NS::computeSpeed(superficial_velocity);
+  }
+
+  ////////////////////////////////////////////////////////////////////
+  ///// Switching across formulation cases
+  ////////////////////////////////////////////////////////////////////
+  if (!_use_superficial)
+  {
+    rho = _epsilon(elem_arg, state) * _rho(elem_arg, state);
+    speed *= (1 / _epsilon(elem_arg, state));
+  }
```

Comment:
I feel like the meaning of `_use_superficial` should be the opposite. If we use it we are providing superficial quantities so we need to have these corrections. Otherwise no corrections are needed.

Reply:
Also speed is already interstitial in the material so if `_speed` is provided then you don't need to do anything here.

Reply:
fixed it